### PR TITLE
Add basic types to the semantics and flattening

### PR DESCRIPTION
**Pull request by Schaeff**
Add a type system with two types:
- FieldElements
- Booleans

FieldElements are used for everything we currently have without types, and booleans are used for IfElse statements.

No changes in the parser, this will be part of a separate PR.

Todo:

- [x] unpack in flattener
- [x] clean up
- [x] figure out how to treat imports
- [x] remove compiled output from integration tests

### CODE REVIEW

**Comment by Schaeff** [comment]
Hey @Kyroy seems like CI is down?

File: src/types/mod.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+impl Type {
+	fn get_constraints<T: Field>(&self) -> Constraints<T> {
+		match self {
+			Type::FieldElement => Constraints::none(),
+			Type::Boolean => Constraints::boolean(),
+		}
+	}
+
```
Comment (JacobEberhardt) [inline src/types/mod.rs]: Maybe add comment:
# of field elements the type maps to.

**Comment by Kyroy** [comment]
@Schaeff Yes it was down. Thanks for the info

File: src/types/mod.rs (hunk 1 above)
Comment (Schaeff) [reply src/types/mod.rs]: Done

**Pull request merged**